swimlane/core/resources/app.py

```diff
@@ -33,8 +33,22 @@ def __init__(self, swimlane, raw):
         self.id = self._raw['id']
         self.tracking_id = self._raw.get('trackingFieldId')
 
-        self._fields_by_id = {f['id']: f for f in self._raw['fields']}
-        self._fields_by_name = {f['name']: f for f in self._raw['fields']}
+        self._fields_by_id = dict()
+        self._fields_by_name = dict()
+        self._defaults = dict()
+
+        for f in self._raw['fields']:
+            self._fields_by_id[f['id']] = f
+            self._fields_by_name[f['name']] = f
+            if f['fieldType'] == "valuesList":
+                t = f['selectionType']
+                for v in f['values']:
+                    if v['selected']:
+                        if t == 'single':
+                            self._defaults[f['name']] = v['name']
+                        elif t == 'multi':
```

Comment:
Is there other choices that you purposely ignored or is it if else statement that can be changed to tertiary? 
= x if this else y?
Also, do we allow more then one default selection for multi? If we do we need to use append, if we don't I suggest putting break statement in the loop to not loop through them all after hitting default, some applications have a lot of values.